contracts/ICO_controller.sol

```diff
@@ -119,30 +123,54 @@ contract ICO_controller is Ownable {
         }
         return true;
     }
+ 
+    function addAirdrop(address[] _airdropAddresses) external onlyOwner returns (bool success) {
+        for(uint i = 0; i < _airdropAddresses.length; i++) {
+            require(_airdropAddresses[i] != address(0));
+            airdropList[_airdropAddresses[i]] = true;
+            totalAirdropAdrresses = totalAirdropAdrresses.add(1);
+        }
+        return true;
+    }
```

Comment:
In this function, it is necessary to check that the address is not yet in the list. Otherwise, it may turn out that the same address will be sent several times and it will be only one in the mapping, but in the totalAirdropAdrresses variable it will be taken into account several times.